src/pluggy/_tracing.py

```diff
@@ -27,26 +26,29 @@ def format_message(self, tags, args):
 
         for name, value in extra.items():
             lines.append("%s    %s: %s\n" % (indent, name, value))
-        return lines
 
-    def processmessage(self, tags, args):
-        if self.writer is not None and args:
-            lines = self.format_message(tags, args)
-            self.writer("".join(lines))
+        return "".join(lines)
+
+    def _processmessage(self, tags, args):
+        if self._writer is not None and args:
+            message = self._format_message(tags, args)
+            self._writer(message)
         try:
-            self._tag2proc[tags](tags, args)
+            processor = self._tags2proc[tags]
```

Comment:
No - the idea is to not over-extend the `try`. Previously, if the `proceessor` function happened to throw a `KeyError`, it will incorrectly be caught and ignored.

Reply:
ahh nice catch!